### ISSUE TITLE
JointPlotVisualizer not poofing

### ISSUE THREAD

**Issue by Naba7** [DistrictDataLabs/yellowbrick#716]
OS:Linux 18.04LTS
Python:3.6.8 Anaconda inc.
Yellowbrick:0.9

So,as said by @rebeccabilbro  I tried to use fit_transform instead of fit in the JointPlotVisualizer,but no change .I am tracing the code along with my terminal output;
CODE:
```python
import pandas as pd
data=pd.read_csv('bikeshare.csv')
X=data[["season","month","hour","holiday","weekday","workingday","weather","temp","feelslike","humidity","windspeed"]]
y=data["riders"]


from yellowbrick.features import JointPlotVisualizer

visualizer=JointPlotVisualizer(feature='temp',target='feelslike')
visualizer.fit_transform(X["temp"],X["feelslike"])
visualizer.poof()
```

OUTPUT:
```
chad7@superuser:~$ cd Desktop/YellowBrick/
chad7@superuser:~/Desktop/YellowBrick$ python3 yb1.py
chad7@superuser:~/Desktop/YellowBrick$ python3 yb1.py
chad7@superuser:~/Desktop/YellowBrick$ 
```
No,changes no figures showing up.

I know I am increasing your tasks with my questions,Sorry!

**Comment by rebeccabilbro** [comment]
Hi again @Naba7 - please try [providing an outpath for your visualizer](http://www.scikit-yb.org/en/latest/faq.html#how-can-i-save-a-yellowbrick-plot):

```python
import pandas as pd
from yellowbrick.features import JointPlotVisualizer

data=pd.read_csv('bikeshare.csv')

X=data[["season","month","hour","holiday","weekday","workingday","weather","temp","feelslike","humidity","windspeed"]]
y=data["riders"]

visualizer=JointPlotVisualizer(feature='temp',target='feelslike')
visualizer.fit_transform(X["temp"],X["feelslike"])
visualizer.poof(outpath="nabas_jointplot.png") # <= provide a path to store the image
```

If what you're looking for is more immediate/interactive viewing of plots, you might try using yellowbrick inside a Jupyter notebook.

Out of curiosity, could you please run the following in a Python REPL and let us know what you see?
```bash
>>> import matplotlib
>>> if not matplotlib.get_backend() == 'Agg':
...     matplotlib.use('Agg', force=True)
>>> print('backend:', matplotlib.get_backend())
```

**Comment by Naba7** [comment]
I had saved it already in outpath and tried,but no success.
the code you had given to run gave output     backend: agg

**Comment by rebeccabilbro** [comment]
@lwgray ☝️?

**Comment by lwgray** [comment]
hi @naba7 

Can you give this a try? by swapping in this line?

visualizer.fit(X['temp'].values, X['feelslike'].values)

**Comment by Naba7** [comment]
okay!

**Comment by Naba7** [comment]
sorry! no success

**Comment by lwgray** [comment]
How about trying...

visualizer.fit(X['temp'].as_matrix(), X['feelslike'].as_matrix())

**Comment by Naba7** [comment]
yb1.py:10: FutureWarning: Method .as_matrix will be removed in a future version. Use .values instead.
  visualizer.fit_transform(X["temp"].as_matrix(),X["feelslike"].as_matrix())


no success.

**Comment by lwgray** [comment]
@naba7 Were you unable to view the jointplot in jupyter notebooks prior to change to ‘.values’ ?

**Comment by Naba7** [comment]
visualizer.fit(X["temp"],X["feelslike"])
visualizer.fit_transform(X["temp"],X["feelslike"])
visualizer.fit_transform(X["temp"].values,X["feelslike"].values)

all,poof the plot without the 45 degree line,in jupyter notebook but not with terminal.

**Comment by lwgray** [comment]
I am stumped @bbengfort any suggestions?

**Comment by Naba7** [comment]
okay! thank you

**Comment by bbengfort** [comment]
Hi @Naba7 I've been going through the quickstart and there does appear to be something odd happening when running it as a script. It does work in a notebook, but the notebook is only rendering the current figure and it does not look like `finalize()` has been called. Possibly there is an issue with `poof()`. I'm looking into it!

**Comment by bbengfort** [comment]
@lwgray @rebeccabilbro found the problem:

The poof method checks to make sure `draw()` has been called on [L209](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/yellowbrick/base.py#L209) by determing if an `ax` has been created and exits if it has not been. The `JointPlotVisualizer` sets up three axes objects, `joint_ax`, `x_ax`, and `y_ax` but never sets `visualizer.ax`. So `poof()` is just exiting silently. 

I've got a PR where I'm converting the quickstart to `.. plot::` - I'll add a fix for this in that PR. 

@Naba7 in the meantime, try the following code as a workaround:

```python
import pandas as pd
import matplotlib.pyplot as plt

from yellowbrick.features import JointPlotVisualizer

data=pd.read_csv('bikeshare.csv')
X=data[["season","month","hour","holiday","weekday","workingday","weather","temp","feelslike","humidity","windspeed"]]
y=data["riders"]

visualizer=JointPlotVisualizer(feature='temp',target='feelslike')
visualizer.fit_transform(X["temp"],X["feelslike"])

visualizer.finalize()
plt.show() # or plt.savefig("myfile.png")
```

**Comment by rebeccabilbro** [comment]
Thanks @bbengfort! Closing this as it will be resolved with #687

**Comment by bbengfort** [comment]
Fixed in #719
